pkg/scheduler/framework/plugins/noderesources/least_allocated.go

```diff
@@ -63,25 +64,46 @@ func (la *LeastAllocated) ScoreExtensions() framework.ScoreExtensions {
 }
 
 // NewLeastAllocated initializes a new plugin and returns it.
-func NewLeastAllocated(_ runtime.Object, h framework.FrameworkHandle) (framework.Plugin, error) {
+func NewLeastAllocated(laArgs runtime.Object, h framework.FrameworkHandle) (framework.Plugin, error) {
+	args := schedulerv1alpha2.RequestedToCapacityRatioArgs{}
+
+	if err := framework.DecodeInto(laArgs, &args); err != nil {
+		return nil, err
+	}
+
+	resToWeightMap := make(resourceToWeightMap)
+	if len(args.Resources) == 0 {
+		// If no resources specified, used the default set.
+		resToWeightMap = defaultRequestedRatioResources
+	}
+	for _, resource := range args.Resources {
+		resToWeightMap[v1.ResourceName(resource.Name)] = resource.Weight
+		if resource.Weight == 0 {
+			// Apply the default weight.
+			resToWeightMap[v1.ResourceName(resource.Name)] = 1
+		}
+	}
+
 	return &LeastAllocated{
 		handle: h,
 		resourceAllocationScorer: resourceAllocationScorer{
 			LeastAllocatedName,
-			leastResourceScorer,
-			defaultRequestedRatioResources,
+			leastResourceScorer(resToWeightMap),
+			resToWeightMap,
 		},
 	}, nil
 }
 
-func leastResourceScorer(requested, allocable resourceToValueMap, includeVolumes bool, requestedVolumes int, allocatableVolumes int) int64 {
-	var nodeScore, weightSum int64
-	for resource, weight := range defaultRequestedRatioResources {
-		resourceScore := leastRequestedScore(requested[resource], allocable[resource])
-		nodeScore += resourceScore * weight
-		weightSum += weight
+func leastResourceScorer(resToWeightMap resourceToWeightMap) func(resourceToValueMap, resourceToValueMap, bool, int, int) int64 {
```

Comment:
what if you make this function a member of `LeastAllocated`?

Reply:
this is coded as is to align with other plugin, if you think make it as a member of `LeatAllocated` bring us some value, we can refactor it with other plugin later.

Reply:
It's just a nit. In fact, `resourceAllocationScorer` should be accepting an interface that this (and other plugins) can implement. This way we avoid the unnecessary closures. But you can leave this for now.